app/controllers/user_profile.rb

```diff
@@ -0,0 +1,46 @@
+class UserProfile < ApplicationController
+  before_action :login_required
+
+  def save_profile_pix(image)
+    unless image.blank?
+      cloud_hash = Cloudinary::Uploader.upload(image)
+      return cloud_hash["url"].split("/").last
+    end
+  end
+
+  def user_info(data, current_user)
+    birthday = get_user_birthday(data[:date])
+    city = current_user.city if data[:city] == "City"
+    state = current_user.state if data[:state] == "State"
+    gender = current_user.gender if data[:gender] == "Gender"
+    birthday = current_user.birthday if birthday.nil?
+    {
+      phone: data[:phone],
+      street_address: data[:street_address],
+      city: city ? city : data[:city],
+      state: state ? state : data[:state],
+      gender: gender ? gender : data[:gender],
+      birthday: birthday,
+      updated_at: DateTime.now
```

Comment:
Do not use `DateTime.now` without zone. Use one of `Time.zone.now`, `DateTime.now.current`, `DateTime.now.in_time_zone`, `DateTime.now.utc`, `DateTime.now.getlocal`, `DateTime.now.iso8601`, `DateTime.now.jisx0301`, `DateTime.now.rfc3339`, `DateTime.now.to_i`, `DateTime.now.to_f` instead.